packages/dx-react-grid/src/plugins/column-order-state.test.jsx

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import { mount } from 'enzyme';
+
+import { Template, Getter, PluginHost } from '@devexpress/dx-react-core';
+
+import { ColumnOrderState } from './column-order-state';
+
+describe('ColumnOrderState', () => {
+  it('should apply initial columns order specified in the "defaultOrder" property in uncontrolled mode', () => {
+    const columnsMock = jest.fn();
+    mount(
+      <PluginHost>
+        <Getter
+          name="columns"
+          value={[{ name: 'a' }, { name: 'b' }]}
+        />
+        <ColumnOrderState
+          defaultOrder={['b', 'a']}
+        />
+        <Template
+          name="root"
+          connectGetters={getter => columnsMock(getter('columns'))}
+        />
+      </PluginHost>,
+    );
+
+    expect(columnsMock.mock.calls[0][0])
+      .toEqual([{ name: 'b' }, { name: 'a' }]);
+  });
+
+  it('should apply columns order according to the "order" property in controlled mode', () => {
+    const columnsMock = jest.fn();
+    mount(
+      <PluginHost>
+        <Getter
+          name="columns"
+          value={[{ name: 'a' }, { name: 'b' }]}
+        />
+        <ColumnOrderState
+          order={['b', 'a']}
+        />
+        <Template
+          name="root"
+          connectGetters={getter => columnsMock(getter('columns'))}
+        />
+      </PluginHost>,
+    );
+
+    expect(columnsMock.mock.calls[0][0])
+      .toEqual([{ name: 'b' }, { name: 'a' }]);
+  });
+
+  it('should change columns order in uncontrolled mode after the "setColumnOrder" action was fired', () => {
+    const columnsMock = jest.fn();
+    const tree = mount(
+      <PluginHost>
+        <Getter
+          name="columns"
+          value={[{ name: 'a' }, { name: 'b' }]}
+        />
+        <ColumnOrderState
+          defaultOrder={['a', 'b']}
+        />
+        <Template
+          name="root"
+          connectGetters={getter => ({ mock: columnsMock(getter('columns')) })}
+          connectActions={action => ({ setColumnOrder: action('setColumnOrder') })}
+        >
+          {({ setColumnOrder }) => (
+            <button
```

Comment:
Not sure if the button required here. Can't we just save setColumnOrder in a local variable and call it instead of simulating a button click?